Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt

```diff
@@ -200,53 +207,82 @@ class FacilityInfoPage : BaseActivity() {
-                    var isOpen: Boolean = false
-                    var openUntil: Long = -1
-                    var opensNext: Long = -1
-
-                    val currentTime = System.currentTimeMillis() / 1000L
-                    for(i in 0 until facilityHoursList.size - 1) {
-                        // Current Time falls into one of the open time slots
-                        if (currentTime >= facilityHoursList[i].first && currentTime < facilityHoursList[i].second) {
-                            isOpen = true
-                            openUntil = facilityHoursList[i].second
-                        }
-                        // Current Time before first time slot of day
-                        else if (i == 0 && currentTime < facilityHoursList[i].first) {
-                            opensNext = facilityHoursList[i].first
-                        }
-                        // Current Time is after the last time slot of day
-                        else if(i == facilityHoursList.size - 2 && currentTime >= facilityHoursList[i].second) {
-                            opensNext = facilityHoursList[facilityHoursList.size - 1].first
-                        }
-                        // Current Time is between two time slots in day
-                        else if(i > 0 && currentTime >= facilityHoursList[i-1].second && currentTime < facilityHoursList[i].first) {
-                            opensNext = facilityHoursList[i].first
-                        }
-                    }
-                    if(opensNext == -1L && openUntil == -1L) {
-                        topBar.setSubtitleTextColor(getResources().getColor(R.color.closed_facility))
-                        topBar.subtitle = "Closed"
-                    }
-                    else {
-                        if(isOpen) {
-                            topBar.setSubtitleTextColor(getResources().getColor(R.color.open_facility))
-                            topBar.subtitle = "Open" + " until " + FluxUtil.parseTime(openUntil)
-
-                        }
-                        else {
-                            topBar.setSubtitleTextColor(getResources().getColor(R.color.closed_facility))
-                            topBar.subtitle = "Closed" + "   opens at " + FluxUtil.parseTime(opensNext)
-                        }
-                    }
+    private fun fetchOperatingHours(date: Date) {
+        val c = Calendar.getInstance()
+        c.time = date
+        c.add(Calendar.DATE, 1)
+        val nextDay: Date = c.time
+        api.facilityHours(facilityId=facilityClass!!.id, startDate = FluxUtil.convertDateObjectToString(date),
+                endDate = FluxUtil.convertDateObjectToString(nextDay),
+        facilityHoursTimeStampsOnResponse = {
+            hoursTimeStampsList ->
+            opHoursTimestamps = hoursTimeStampsList
+        },
+        facilityHoursStringsOnResponse = {
+            hoursStringsList ->
+            opHoursStrings = hoursStringsList
+        })
```

Comment:
will things get messed up if `opHoursStrings` is not set in time for the UI updates?

Reply:
I guess it won't throw an error since it's just an empty list...